src/generators/javascript/presets/CommonPreset.ts

```diff
@@ -0,0 +1,213 @@
+}`;
+  });
+  return unmarshalProperties.join('\n');
+}
+
+function renderUnmarshalPatternProperties(model: CommonModel, renderer: JavaScriptRenderer, inputModel: CommonInputModel) {
+  let unmarshalPatternProperties = '';
+  let setPatternPropertiesMap = '';
+  if (model.patternProperties !== undefined) {
+    for (const [pattern, patternModel] of Object.entries(model.patternProperties)) {
+      let patternPropertyName = getUniquePropertyName(model, `${pattern}${DefaultPropertyNames.patternProperties}`);
+      patternPropertyName = renderer.nameProperty(patternPropertyName, patternModel);
+      const modelInstanceVariable = 'value as any';
+      const unmarshalCode = renderUnmarshalProperty(modelInstanceVariable, patternModel, inputModel, renderer);
+      setPatternPropertiesMap += `if (instance.${patternPropertyName} === undefined) {instance.${patternPropertyName} = new Map();}\n`;
+      unmarshalPatternProperties += `//Check all pattern properties
+if (key.match(new RegExp('${pattern}'))) {
+  instance.${patternPropertyName}.set(key, ${unmarshalCode});
+  continue;
+}`;
+    }
+  }
+  return { unmarshalPatternProperties, setPatternPropertiesMap };
+}
+
+function renderUnmarshalAdditionalProperties(model: CommonModel, renderer: JavaScriptRenderer, inputModel: CommonInputModel) {
+  let unmarshalAdditionalProperties = '';
+  let setAdditionalPropertiesMap = '';
+  if (model.additionalProperties !== undefined) {
+    let additionalPropertyName = getUniquePropertyName(model, DefaultPropertyNames.additionalProperties);
+    additionalPropertyName = renderer.nameProperty(additionalPropertyName, model.additionalProperties);
+    const modelInstanceVariable = 'value as any';
+    const unmarshalCode = renderUnmarshalProperty(modelInstanceVariable, model.additionalProperties, inputModel, renderer);
+    setAdditionalPropertiesMap = `if (instance.${additionalPropertyName} === undefined) {instance.${additionalPropertyName} = new Map();}`;
+    unmarshalAdditionalProperties = `instance.${additionalPropertyName}.set(key, ${unmarshalCode});`;
+  }
+  return { unmarshalAdditionalProperties, setAdditionalPropertiesMap };
+}
+
+/**
+ * Render `unmarshal` function based on model
+ */
+function renderUnmarshal({ renderer, model, inputModel }: {
+  renderer: JavaScriptRenderer,
+  model: CommonModel,
+  inputModel: CommonInputModel
+}): string {
+  const properties = model.properties || {};
+  const { unmarshalPatternProperties, setPatternPropertiesMap } = renderUnmarshalPatternProperties(model, renderer, inputModel);
+  const { unmarshalAdditionalProperties, setAdditionalPropertiesMap } = renderUnmarshalAdditionalProperties(model, renderer, inputModel);
+  const unmarshalProperties = renderUnmarshalProperties(model, renderer, inputModel);
+  const formattedModelName = renderer.nameType(model.$id);
+  const propertyNames = Object.keys(properties).map((prop => `"${prop}"`));
+  return `public static unmarshal(json: string | object): ${formattedModelName} {
```

Comment:
Don't need the types here 🙂

Reply:
I got errors as initially I haven't used type for variable then I added type and it got resolved.

Reply:
Can you try adding the test file so I can see the output and what it complains about? 🙂

Reply:
Done, please review.